src/addr.rs

```diff
@@ -70,13 +70,8 @@ impl VirtAddr {
     /// This function performs sign extension of bit 47 to make the address canonical, so
     /// bits 48 to 64 are overwritten. If you want to check that these bits contain no data,
     /// use `new` or `try_new`.
-    pub fn new_unchecked(mut addr: u64) -> VirtAddr {
-        if addr.get_bit(47) {
-            addr.set_bits(48..64, 0xffff);
-        } else {
-            addr.set_bits(48..64, 0);
-        }
-        VirtAddr(addr)
+    pub const fn new_unchecked(addr: u64) -> VirtAddr {
+        VirtAddr((addr.wrapping_mul(0x1_0000) as i64 / 0x1_0000) as u64)
```

Comment:
Nice trick! Since this is less apparent as the old implementation, could you add a comment to explain how this works?

Reply:
Sure. Done.